packages/odata-service-inquirer/src/prompts/datasources/metadata-file/validators.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Validates the metadata file, returning an error string, if not valid, or an object with the valid metadata and version.
+ *
+ * @param path
+ * @param odataVersion
+ * @returns
+ */
+
+import { OdataVersion } from '@sap-ux/odata-service-writer';
+import { readFile } from 'fs/promises';
+import { validateODataVersion } from '../../validators';
+import { t } from '../../../i18n';
+
+export const validateMetadataFile = async (
+    path: string,
+    odataVersion?: OdataVersion
+): Promise<
+    | {
+          version: OdataVersion;
+          metadata: string;
+      }
+    | string
+    | boolean
+> => {
+    if (!path) {
+        return false;
+    }
+
+    try {
+        const metadata = await readFile(path, 'utf-8');
```

Comment:
File will exist before loading so no dependency on `this.fs` from the generator, adding this note incase I review it again and forget!